added arabic translation guidelines (#62166)
apache/airflow: .github/skills/airflow-translations/locales/ar.md

 <!-- SPDX-License-Identifier: Apache-2.0
      https://www.apache.org/licenses/LICENSE-2.0 -->

# Arabic (ar) Translation Guidelines for Apache Airflow

This document provides Arabic-specific guidelines for translating Apache Airflow terminology and documentation.

## Terms to Keep in English

> The following technical terms should **remain in English** in Arabic translations:

Reviewer: Not coherent with the instructions below (Only Dag and XCom remain in English, the other should be translated).

### Core Technical Terms (مصطلحات تقنية أساسية)

> - **DAG** (Directed Acyclic Graph) - Keep as "DAG"

Reviewer: ```suggestion
- **DAG/Dag** (Directed Acyclic Graph) - Keep as "Dag"
```

- **XCom** (Cross-Communication) - Keep as "XCom"

> - **Operator** - Keep as "Operator" (المُشغِّل) or in English

Reviewer: ```suggestion
- **Operator** - Translate as "Operator" (المُشغِّل)
```

> - **Task Instance** - Keep as "Task Instance" or translate as "مثيل المهمة"

Reviewer: ```suggestion
- **Task Instance** - Translate as "مثيل المهمة"
```

> - **Pool** - Keep as "Pool" (المجموعة) or in English

Reviewer: ```suggestion
- **Pool** - Translate as المجموعة
```

> - **Hook** - Keep as "Hook" or translate as "خطاف"
> - **Sensor** - Keep as "Sensor" (المستشعر) or in English

Reviewer: ```suggestion
```
Currently we don't use these terms in i18n translation files

### UI Components (مكونات واجهة المستخدم)

> - **Tree View** - Keep as "Tree View" or translate as "عرض الشجرة"

Reviewer: ```suggestion
```
Currently not in use

> - **Graph View** - Keep as "Graph View" or translate as "عرض المخطط"

Reviewer: ```suggestion
- **Graph View** - Translate as "عرض الرسم"
```

> - **Gantt Chart** - Keep as "Gantt Chart" or translate as "مخطط جانت"

Reviewer: ```suggestion
- **Gantt Chart** - Translate as "مخطط جانت"
```

> - **Grid View** - Keep as "Grid View" or translate as "عرض الشبكة"

Reviewer: ```suggestion
- **Grid View** - Translate as "عرض الشبكة"
```

### Code and Technical References

- All Python class names, function names, and variables
- Configuration keys (e.g., `dag_id`, `task_id`)
- Command-line arguments and flags
- File paths and URLs

## Common Airflow Terms in Arabic

### Core Concepts (المفاهيم الأساسية)

> | English | Arabic | Notes |

Reviewer: When we use a native Arabic word for a term, noting its grammatical gender is redundant - please remove where applicable (same comment applies for all tables)

|---------|--------|-------|
| Task | مهمة | Feminine noun (مهمة) |

> | DAG | DAG | Keep in English |

Reviewer: ```suggestion
| Dag | Dag| Keep in English |
```

| XCom | XCom | Keep in English |

> | Workflow | سير العمل | Masculine noun |
> | Pipeline | خط أنابيب | Masculine noun |

Reviewer: ```suggestion
```
Currently not translated (I'd rather wait for an actual usage, an then we'll decide how to translate)

| Dataset | مجموعة بيانات | Feminine noun |
| Connection | اتصال | Masculine noun |
| Variable | متغير | Masculine noun |

> | Provider | موفر | Masculine noun |

Reviewer: ```suggestion
| Provider | حُزْمَة | Masculine noun |
```

> | Trigger | تشغيل / مُشغِّل | Can be verb or noun |

Reviewer: Please note below that we use the term مُطلِق for trigger**er**, to avoid confusion

> | Scheduler | الجدولة / المجدول | For scheduler component |

Reviewer: ```suggestion
| Scheduler | المُجَدْوِل | For scheduler component |
```

> | Executor | المنفذ / المُنفِّذ | For executor component |

Reviewer: ```suggestion
| Executor | منفذ | For executor component |
```

| Worker | العامل | For worker process |
| Webserver | خادم الويب | Standard translation |
| Database | قاعدة البيانات | Standard translation |

> ### Actions (الإجراءات)

Reviewer: Basically it just duplicates the actions in dags.json (there's one hallucinated state, like upstream_skipped). 
Please ensure that terms exactly match, or just refer to this file in a sentence.

| English | Arabic | Notes |
|---------|--------|-------|
| Run | تشغيل | Verb: يشغِّل، تشغيل |
| Execute | تنفيذ | Verb: ينفِّذ، تنفيذ |
| Trigger | تشغيل | Verb: يشغِّل، تشغيل |
| Clear | مسح | Verb: يمسح، مسح |
| Retry | إعادة المحاولة | Noun phrase |

> | Retry | أعاد المحاولة | Verb phrase |

Reviewer: ```suggestion
```
Redundant

| Fail | فشل | Verb: يفشل، فشل |

> | Mark as Failed | تعليم كمفشل | Phrase |

Reviewer: ```suggestion
```
Redundant

| Success | نجح | Verb: ينجح، نجاح |

> | Mark as Success | تعليم كناجح | Phrase |

Reviewer: ```suggestion
```
Redundant

| Pause | إيقاف مؤقت | Verb: يوقف مؤقتاً |
| Unpause | إلغاء الإيقاف المؤقت | Verb phrase |

### States (الحالات)

| English | Arabic | Notes |
|---------|--------|-------|
| success | ناجح | Adjective: ناجح (masc.), ناجحة (fem.) |
| running | قيد التشغيل | Phrase |
| failed | فاشل | Adjective: فاشل (masc.), فاشلة (fem.) |
| upstream_failed | فشل المنبع | Phrase |
| skipped | تم التخطي | Phrase |
| upstream_skipped | تم تخطي المنبع | Phrase |
| queued | في قائمة الانتظار | Phrase |
| scheduled | مجدول | Adjective |
| removed | تمت الإزالة | Phrase |

> | deferred | مؤجل | Adjective |

Reviewer: Basically it just duplicates the states in common.json (there's one hallucinated state, like upstream_skipped). Please ensure that terms exactly match.

## Pluralization Patterns in Arabic

> Arabic has complex pluralization rules with different forms:

Reviewer: 1. I'm a bit doubtful regarding the added value of the instructions here, as complex rules (which indeed they are in Arabic) cannot be oversimplified with a number of examples - it's either that the model knows them  beforehand or not.

2. Instead:
a. I'd mention that it should consider the known pluralization rules in Arabic plus specific forms in the context of Airflow. For example - keeping "2 Dags" as "2 Dags" and not mixing it up as "Dagان/ Dagين".
b. I would give an example of the the **format** of the pluralization (maybe combine with the paragraph below):
```json
    "noPoolsFound": "لا توجد مجموعات موارد متاحة",
    "pool_few": "مجموعات",
    "pool_many": "مجموعة",
    "pool_one": "مجموعة",
    "pool_other": "مجموعة",
    "pool_two": "مجموعتان",
    "pool_zero": "لا يوجد أي مجموعة",
 ```

### Dual Form (المثنى) - for exactly 2

Used with the suffix "ـان" (ān) or "ـين" (ayn):

> - مهمة → مهمتان (2 tasks)

Reviewer: ```suggestion
- مهمة → مهمتـين/مهمتان (2 tasks)
```

> - DAG → 2 DAGs (keep English with number)

Reviewer: ```suggestion
- Dag → 2 Dags (keep English with number)
```

### Sound Masculine Plural (جمع المذكر السالم)

Suffix "ـون" (ūn) in nominative, "ـين" (īn) in accusative/genitive:

- موفر → موفرون (providers)
- عامل → عمالون (workers)
- متغير → متغيرون (variables)

### Sound Feminine Plural (جمع المؤنث السالم)

Suffix "ـات" (āt):

- مهمة → مهمات (tasks)
- مجموعة → مجموعات (datasets/groups)
- اتصال → اتصالات (connections)

### Broken Plural (الجمع التكسيري)

Irregular patterns:

- خط أنابيب → خطوط أنابيب (pipelines)
- قاعدة → قواعد (databases)

### Using Numerals with Arabic

In Airflow UI and messages, numerals are typically formatted as:

- **Singular**: 1 مهمة (1 task)
- **Dual**: 2 مهمتان (2 tasks) or 2 DAGs
- **Plural (3-10)**: 5 مهام (5 tasks)
- **Plural (11+)**: 15 مهمة (15 tasks)

Note: For numbers 3-10, the noun is typically in plural form. For numbers 11+, the noun uses singular form.

## RTL (Right-to-Left) Considerations

1. **Direction**: Arabic text is written from right to left (RTL)
2. **Mixed Content**: When mixing Arabic with English terms, maintain proper spacing
3. **Punctuation**: Periods, commas, and other punctuation should be placed appropriately
4. **Code Blocks**: Keep code blocks in English/LTR direction

> 5. **Numbers**: Numbers (0-9) are written left-to-right even in Arabic text

Reviewer: Please mention explicitly that we only use Arabic numrals (1,2,3...) and not Eastern-Arabic numerals (٣, ٢, ١...)

Example:

```text
تم تشغيل DAG بنجاح (DAG run successful)
```

## Gender Agreement

Arabic nouns have grammatical gender that affects adjectives and verbs:

- **Task (مهمة)**: Feminine
  - Singular: مهمة واحدة (one task)

>   - Plural: مهمات متعددة (multiple tasks)

Reviewer: Now sure regarding the added value

> - **DAG**: Keep in English, treat as masculine or feminine based on context (لفظ "مخطط" is masculine)

Reviewer: ```suggestion
```
Redundant, already mentioned above

- **Pipeline (خط أنابيب)**: Masculine
- **Connection (اتصال)**: Masculine

## Translation Style Guidelines

### 1. Technical Terminology

> - Keep technical terms like DAG, XCom, Operator in English when:

Reviewer: ```suggestion
- When encountering an English technical English term which is not already mentioned above, keep technical terms in English when:
```

  - They appear in code or configuration examples
  - No clear Arabic equivalent exists
  - The term is widely used in English in the technical community

### 2. UI Labels

- Keep UI labels concise and consistent
- Use standard Arabic technical translations where available

> - Example: "Tree View" → "عرض الشجرة" or keep "Tree View"

Reviewer: What are UI labels? I'd remove it

### 3. Verbs and Actions

- Use imperative form for buttons and actions:
  - "Run" → "شغِّل" (imperative)
  - "Clear" → "امسح" (imperative)

>   - "Trigger" → "شغِّل" (imperative)

Reviewer: We don't use the imperative as it is less natural in Arabic.
We use verbal nouns, examples are in the code.

### 4. Documentation

- Use formal Modern Standard Arabic (MSA)
- Maintain consistency with terminology throughout
- Provide English terms in parentheses when introducing new technical terms

### 5. Error Messages

- Keep error messages clear and actionable
- Include technical details in English when necessary
- Example: "فشل تشغيل المهمة: Task instance not found"

> ## Common Translation Patterns

Reviewer: Please take examples directly from the code, some phrases here are hallucinated
